baseband/vlbi_base/payload.py

```diff
@@ -98,7 +100,12 @@ def fromdata(cls, data, bps=2):
             Number of bits per sample to use (for complex data, for real and
             imaginary part separately; default: 2).
         """
-        sample_shape = data.shape[1:]
+        try:
+            sample_shape = cls._sample_shape_cls(*data.shape[1:])
```

Comment:
This is why I thought you didn't need to initialize using `_sample_shape_cls` in the subclasses, but I now see it is just for `fromdata`. If we move this to `__init__`, it may not be necessary to do anything special here.

Reply:
I'm not sure I understand - are you recommending that we move the try/except block to `__init__` (and pass `data.shape[1:]` as a tuple to it in `fromdata`).  I'm not sure that makes sense since the try/except is never used except in `fromdata`, and sample shapes are already properly initialized in format subclasses.

Reply:
With the above, passing `data.shape[1:]` will work fine.